lib/spack/spack/cmd/create.py

```diff
@@ -637,6 +645,31 @@ def setup_parser(subparser):
     subparser.add_argument(
         '-b', '--batch', action='store_true',
         help="don't ask which versions to checksum")
+    subparser.add_argument(
+        '-g', '--git', action='store', nargs='?', default='#AUTO-GIT-URL#',
+        help="use git to download source from repository passed in url argument")
+    subparser.add_argument(
+        '-V', '--version',
+        help='override derived package version')
+    subparser.add_argument('-B', '--branch',
+                           help="specify branch(es) of git repository. "
+                                "Separate multiple branches with space. "
+                                "Not guaranteed to be reproducible, use "
+                                "`--commit` when possible. "
+                                "Only used for git URLs.",
+                           nargs='+')
+    subparser.add_argument('-T', '--tag',
+                           help="specify tag(s) of git repository. "
+                                "Separate multiple tags with space. "
+                                "Not guaranteed to be reproducible, "
+                                "use `--commit` when possible. "
+                                "Only used for git URLs.",
```

Comment:
This guidance doesn't apply to the `tags` option:

```suggestion
                                "Separate multiple tags with space.",
```